core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store_test.go

```diff
@@ -0,0 +1,159 @@
+package evm
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+var (
+	upkeepId1 = big.NewInt(100)
+	upkeepId2 = big.NewInt(200)
+	trigger1  = ocr2keepers.Trigger{
+		BlockNumber: 95,
+		BlockHash:   "0x1231eqwe12eqwd",
+	}
+	trigger2 = ocr2keepers.Trigger{
+		BlockNumber: 125,
+		BlockHash:   "0x1231eqwe12eqwd",
+	}
+	payload2 = ocr2keepers.NewUpkeepPayload(upkeepId1, ConditionalType, BlockKey2, trigger1, []byte{})
+	payload3 = ocr2keepers.NewUpkeepPayload(upkeepId2, LogTriggerType, BlockKey2, trigger1, []byte{})
+	payload4 = ocr2keepers.NewUpkeepPayload(upkeepId1, LogTriggerType, BlockKey1, trigger2, []byte{})
+	payload5 = ocr2keepers.NewUpkeepPayload(upkeepId1, LogTriggerType, BlockKey3, trigger1, []byte{})
+)
+
+const (
+	ConditionalType = 0
+	LogTriggerType  = 1
+	Block1          = 111
+	Block3          = 113
+	BlockKey1       = "111|0x123123132132"
+	BlockKey2       = "112|0x565456465465"
+	BlockKey3       = "113|0x111423246546"
+	InvalidBlockKey = "2220x565456465465"
```

Comment:
nitpick but aren't these exported now since they're capitalized? do we need to export these?